adi/adpd1080.py

```diff
@@ -0,0 +1,110 @@
+#       the documentation and/or other materials provided with the
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class adpd1080(rx, context_manager):
+
+    """ADPD1080 photo-electronic device."""
+
+    channel = []  # type: ignore
+    _device_name = ""
+    _rx_data_type = ["<u4", "<u4", "<u4", "<u4", "<u4", "<u4", "<u4", "<u4"]
+
+    def __init__(self, uri="", device_index=0):
+        """ADPD1080 class constructor."""
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = ["adpd1080"]
+
+        self._ctrl = None
+        index = 0
+
+        # Selecting the device_index-th device from the adpd188 family as working device.
+        for device in self._ctx.devices:
+            if device.name in compatible_parts:
+                if index == device_index:
+                    self._ctrl = device
+                    self._rxadc = device
+                    break
+                else:
+                    index += 1
+
+        # dynamically get channels and sorting them after the color
+        # self._ctrl.channels.sort(key=lambda x: str(x.id[14:]))
+
+        for ch in self._ctrl._channels:
+            name = ch._id
+            self._rx_channel_names.append(name)
+            self.channel.append(self._channel(self._ctrl, name))
+        rx.__init__(self)
+        self.rx_buffer_size = 16
+
+    def rx(self):
+            buff = super().rx()
+            del self._rx__rxbuf
+            return buff
```

Comment:
Do you really need to delete the buffer between calls?

Reply:
This was done because in an application we needed to read a few samples and then take the average and write it into the offset register of the device. But writes to the device did not work anymore after buffer reads, unless the buffer was deleted first.

Reply:
Hmm that seems really strange. Is this a bug in the driver or the desired behavior?

Reply:
I can leave it up to you and @scuciurean about if this is proper. Just let me know and we'll merge.